treeherder/webapp/api/intermittents_view.py

```diff
@@ -44,22 +46,36 @@ class FailuresByBug(generics.ListAPIView):
     def get_queryset(self):
         startday = self.request.query_params.get('startday').encode('utf-8') + ' 00:00:00'
         endday = get_end_of_day(self.request.query_params.get('endday').encode('utf-8'))
-        tree = get_tree(self.request.query_params.get('tree').encode('utf-8'))
+        tree = list(get_tree(self.request.query_params.get('tree').encode('utf-8')))
         bug_id = int(self.request.query_params.get('bug'))
 
-        queryset = BugJobMap.objects.select_related('job', 'push').filter(bug_id=bug_id, job__repository_id__in=tree,
-                                                                          job__push__time__range=(startday, endday))\
-            .values('bug_id', 'job_id', 'job__push__time', 'job__repository__name', 'job__option_collection_hash',
-                    'job__signature__job_type_name', 'job__push__revision', 'job__machine_platform__platform')\
-            .order_by('-job__push__time')
+        queryset = BugJobMap.objects.filter(bug_id=bug_id,
+                                            job__repository_id__in=tree,
+                                            job__push__time__range=(startday, endday)
+                                            ).select_related('job', 'push').values(
+                                            'bug_id', 'job_id', 'job__push__time', 'job__machine_platform__platform',
+                                            'job__repository__name', 'job__push__revision',
+                                            'job__signature__job_type_name', 'job__option_collection_hash',
+                                             ).order_by('-job__push__time')
+
+        hash_list = []
 
         for item in queryset:
             test_type = item['job__signature__job_type_name']
             item['test_suite'] = re.sub(r'.+/', '', test_type)
+            match = filter(lambda x: item['job__option_collection_hash'] == x, hash_list)
+            if len(match) == 0:
+                hash_list.append(item['job__option_collection_hash'])
+
+        hash_query = OptionCollection.objects.filter(option_collection_hash__in=hash_list).select_related('option')\
```

Comment:
nit: another one of those line continuations we could change to parameter wrapping